app/models/rule_reference.rb

```diff
@@ -18,16 +18,29 @@ class RuleReference < ApplicationRecord
                      message: 'and label combination already taken'
                    }
 
-  def self.find_from_oscap(oscap_references)
-    oscap_references.inject(where('1=0')) do |rel, reference|
-      rel.or(where(href: reference.href, label: reference.label))
+  class << self
+    def find_from_oscap(oscap_references)
+      oscap_references.inject(where('1=0')) do |rel, reference|
+        rel.or(where(href: reference.href, label: reference.label))
+      end
```

Comment:
@dLobatog this is the `inject` style query to find all of a collection of objects from the database.